fv3core/stencils/delnflux.py

```diff
@@ -1,82 +1,903 @@
-from gt4py.gtscript import PARALLEL, computation, interval
+from typing import Optional
+
+import gt4py.gtscript as gtscript
+import numpy as np
+from gt4py.gtscript import (
+    __INLINED,
+    FORWARD,
+    PARALLEL,
+    computation,
+    horizontal,
+    interval,
+    region,
+)
 
 import fv3core._config as spec
-import fv3core.utils.corners as corners
+import fv3core.utils.global_config as global_config
 import fv3core.utils.gt4py_utils as utils
 from fv3core.decorators import gtstencil
-from fv3core.stencils.basic_operations import copy_stencil
-from fv3core.utils.typing import FloatField, FloatFieldIJ
+from fv3core.utils.grid import axis_offsets
+from fv3core.utils.typing import FloatField, FloatFieldIJ, FloatFieldK
+
+
+def copy_corners_x_nord(q: FloatField):
```

Comment:
Tests indicate we need to split these into multiple stencils to prevent template depth errors with the GT backends, something like:
```python
conditional_corner_copy_x_funcs = [
        copy_corners_x_nord0,
        copy_corners_x_nord1,
        copy_corners_x_nord2,
        copy_corners_x_nord3,
    ]
    conditional_corner_copy_x_stencils = [
        StencilWrapper(
            conditional_corner_copy_x_func,
            externals=full_ax_offsets,
        )
        for conditional_corner_copy_x_func in conditional_corner_copy_x_funcs
    ]
...
    for i, conditional_corner_copy_x in enumerate(conditional_corner_copy_x_stencils):
        if nord[i] > 0:
            conditional_corner_copy_x(
                d2, origin=(grid.isd, grid.jsd, 0), domain=(grid.nid, grid.njd, nk)
            )
```